### PR TITLE
Apply filters at start

### LINKED ISSUE #509: On launch, "Show unavailable games" filter is checked, but not in effect.
[Using this screenshot as a reference](http://i.imgur.com/Y0pPNOi.png), you can see that the "Show unavailable games" checkbox is enabled, but the games list doesn't show unavailable games (games in progress with no empty slots).

Unchecking the box and hitting "Ok", then checking the box and repeating will fix this behavior, but if it is checked by default, the behavior should be in effect at that point.

### PR DESCRIPTION
**Pull request by kaiserfro**
Fixes #509 

Basically, this patch refactors the filter code such that persistence of the filters is not done by the dialog class, but the model instead.  Then the dialog uses the model to set the state of the dialog properly, then save the data back to the model, then on to the qsettings.

I added a load filters function that is called at the time the games window is created that sets the filters up initially from the saved settings.

### CODE REVIEW

File: cockatrice/src/dlg_filter_games.h [hunk 1]
```diff
@@ -21,17 +22,14 @@ class DlgFilterGames : public QDialog {
     QSpinBox *maxPlayersFilterMaxSpinBox;
 
     const QMap<int, QString> &allGameTypes;
+    // This needs a const someplace
```
Comment (Daenyth) [inline cockatrice/src/dlg_filter_games.h]: I'd like to see the comment explain why, or it's not very useful

File: cockatrice/src/gamesmodel.cpp [hunk 2]
```diff
@@ -221,12 +223,62 @@ void GamesProxyModel::resetFilterParameters()
...
+{
+    QSettings settings;
+    settings.beginGroup("filter_games");
+
+    unavailableGamesVisible = settings.value("unavailable_games_visible", false).toBool();
+    passwordProtectedGamesVisible = settings.value("password_protected_games_visible", false).toBool();
+    gameNameFilter = settings.value("game_name_filter", "").toString();
+    creatorNameFilter = settings.value("creator_name_filter", "").toString();
+    maxPlayersFilterMin = settings.value("min_players", 1).toInt();
+    maxPlayersFilterMax = settings.value("max_players", 99).toInt();
```
Comment (Daenyth) [inline cockatrice/src/gamesmodel.cpp]: I'd like to see the number here reused from a constant

**Comment by Daenyth** [comment]
Just to confirm, test case that fails prior to this and passes after:

save filter, close cockatrice, open cockatrice, view games list - games should be filtered

**Comment by kaiserfro** [comment]
Ok, i pushed with the couple of fixes.

Yes, this change should apply the filters after a restart or log out, then back in.  When i put this together, there was a question in my mind as to whether or not all of the filters should be reapplied upon start or not.  I did them all, even though, i think the creator name, for instance is less useful as a persistent filter.

**Comment by kaiserfro** [comment]
This fixes #360 also.

**Comment by ctrlaltca** [comment]
Code seems fine and works as expected. :+1:

**Comment by ZeldaZach** [comment]
I see no issues in this when testing

**Pull request merged**